module-system/sov-modules-api/src/lib.rs

```diff
@@ -265,3 +275,44 @@ pub trait RpcRunner {
     type Context: Context;
     fn get_storage(&self) -> <Self::Context as Spec>::Storage;
 }
+
+/// Sorts ModuleInfo objects by their dependencies
+pub fn sort_by_dependencies<C: Context>(
+    modules: Vec<&dyn ModuleInfo<Context = C>>,
+) -> Result<Vec<&dyn ModuleInfo<Context = C>>, anyhow::Error> {
+
+    fn visit_module<'a, C: Context>(
+        module: &'a dyn ModuleInfo<Context = C>,
+        visited: &mut HashSet<&'a C::Address>,
+        sorted_modules: &mut std::vec::Vec<&'a (dyn ModuleInfo<Context = C> + 'a)>,
+        module_map: &HashMap<&<C as Spec>::Address, &'a (dyn ModuleInfo<Context = C> + 'a)>
+    ) -> Result<(), anyhow::Error> {
```

Comment:
let's return `syn::Error` and produce a nice compile error if something goes wrong.